src/main/java/com/acrylic/animatedmc/math/Line.java

```diff
@@ -0,0 +1,70 @@
+package com.acrylic.animatedmc.math;
+
+import org.bukkit.Location;
+
+public class Line implements Geometry {
+
+    private Location origin;
+    private int interpolationPoints = 10;
+    private double deltaX;
+    private double deltaY;
+    private double deltaZ;
+
+    public void setOrigin(Location origin) {
+        this.origin = origin;
+    }
+
+    public void setToLocation(Location location) {
+        Location difference = location.clone().subtract(origin);
+        this.deltaX = difference.getX();
+        this.deltaY = difference.getY();
+        this.deltaZ = difference.getZ();
+    }
+
+    @Override
+    public Location getOrigin() {
+        return this.origin;
+    }
+
+    @Override
+    public Location step(int index) {
+        double percent = index / (double) interpolationPoints;
+        return new Location(origin.getWorld(),
+                origin.getX() + (deltaX * percent),
+                origin.getY() + (deltaY * percent),
+                origin.getZ() + (deltaZ * percent)
```

Comment:
Recommended to just use index. Delta values should be the change in X,Y,Z values for each point.

Reply:
aight